### PR TITLE
Interpreter configuration

### PR DESCRIPTION
**Pull request by elliotmassen**
This PR is dependent on https://github.com/opendialogai/core/pull/486 and exposes an API for managing `ComponentConfiguration`.

### CODE REVIEW

File: app/Http/Controllers/API/ComponentConfigurationController.php [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    }
+
+    /**
+     * Update the specified resource in storage.
+     *
+     * @param ComponentConfigurationRequest $request
+     * @param int $id
+     * @return Response
+     */
+    public function update(ComponentConfigurationRequest $request, int $id)
```
Comment (stu-greenshoots) [inline app/Http/Controllers/API/ComponentConfigurationController.php:66]: Can we use the ConfigurationRequest object in the method signature here to deal with the fetching and 404s for us?

File: app/Http/Requests/ComponentConfigurationRequest.php [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+namespace App\Http\Requests;
+
+use App\Rules\ComponentConfigurationRule;
+use App\Rules\ComponentRegistrationRule;
+use Illuminate\Foundation\Http\FormRequest;
+use Illuminate\Validation\Rule;
+
+class ComponentConfigurationRequest extends FormRequest
+{
+    protected $stopOnFirstFailure = true;
```
Comment (stu-greenshoots) [inline app/Http/Requests/ComponentConfigurationRequest.php:12]: Why do we want to stop on the first failure? Could this potentially lead to a user having to submit multiple times to see all the errors?

**Review by stu-greenshoots — CHANGES_REQUESTED** [review]
Just a couple of small points to look at / clarify

File: app/Http/Requests/ComponentConfigurationRequest.php (hunk 2 above)
Comment (elliotmassen) [reply app/Http/Requests/ComponentConfigurationRequest.php:12]: Good point, I set this to get round a temporary issue I was having, and didn't think to unset it!

File: composer.json [hunk 3]
```diff
@@ -15,7 +15,7 @@
         "laravel/tinker": "^2.0",
         "laravel/ui": "^3.0",
         "maennchen/zipstream-php": "^2.0",
-        "opendialogai/core": "1.x-dev",
+        "opendialogai/core": "dev-feature/OPNDLG-685--interpreter-configuration",
```
Comment (elliotmassen) [inline composer.json:18]: This will need to be updated before merging.

**Pull request merged**